packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -356,6 +359,30 @@ export default function lwc(pluginOptions: RollupLwcOptions = {}): Plugin {
             const rollupMap = map as SourceMapInput;
             return { code, map: rollupMap };
         },
+        api: {
+            updateOptions(options: {
+                rootDir: string;
+                modules?: ModuleRecord[];
+                experimentalComplexExpressions?: boolean;
+                apiVersion?: number;
+            }) {
+                rootDir = options.rootDir;
+
+                modules = [
+                    ...(options.modules ?? []),
+                    ...DEFAULT_MODULES,
+                    { dir: options.rootDir },
+                ];
+
+                if (options.experimentalComplexExpressions !== undefined) {
+                    experimentalComplexExpressions = options.experimentalComplexExpressions;
+                }
+
+                if (options.apiVersion !== undefined) {
+                    apiVersion = options.apiVersion;
+                }
+            },
+        },
```

Comment:
Providing an api to change some options to avoid recreating the plugin for tests and fixtures which require different settings.

I've since found a cleaner approach which I might use as reference: https://github.com/vitejs/vite-plugin-vue/blob/main/packages/plugin-vue/src/index.ts#L192-L200